scripts/update-codegen.sh

```diff
@@ -18,30 +20,18 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-FORCE="${FORCE:-false}"
-[[ -d pkg/generated ]] && ! ${FORCE} && exit 0
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
 
-SCRIPT_ROOT=$(cd `dirname "${BASH_SOURCE[0]}"`/.. && pwd)
-[[ $# -eq 1 ]] && VERIFY_ONLY=true || VERIFY_ONLY=false
-TMP_DEST="${1:-${SCRIPT_ROOT}/_tmp_dest}"
+source "${CODEGEN_PKG}/kube_codegen.sh"
 
-cleanup() {
-  [[ -d "${TMP_DEST}/github.com/skupperproject/skupper" ]] && \
-  ! ${VERIFY_ONLY} && rm -rf "${TMP_DEST}"
-}
-trap "cleanup" EXIT SIGINT
+kube::codegen::gen_helpers \
+    --boilerplate "${SCRIPT_ROOT}/scripts/boilerplate.go.txt" \
+    "${SCRIPT_ROOT}/pkg/apis"
 
-API_VERSION=`grep k8s.io/apimachinery go.mod | awk '{print $NF}'`
-go get -d k8s.io/code-generator@${API_VERSION}
```

Comment:
Maybe this could be kept as an alternative, in case there is not a `./vendor` directory?
After that, we could change it a bit to copy kube-gen.sh to SCRIPT_ROOT, like:
```
cp ${GOPATH}/pkg/mod/k8s.io/code-generator\@${API_VERSION}/kube_codegen.sh ${SCRIPT_ROOT}
```

Reply:
I believe the tools.go causes the code-generator to be considered a dependency, whether vendored or not. I am mostly copying from the 'example'.  The recent change to golang version  means generate-groups.sh is no longer available.

Reply:
... and possibly add vendor/ to .gitignore?

Reply:
I've pushed an update that I believe should find the module if not using vendoring. Also added vendor to .gitignore.